spec/aws.md

```diff
@@ -0,0 +1,56 @@
+AWS
+===
+
+## Naming and metadata conventions
+
+Components of names are:
+
+    org   -> the organisation producing the service, e.g. com.gu
+    stack -> conceptual group the app belongs too
+    app   -> the name of the service itself
+    STAGE -> the environment the app is running in (e.g. PROD), must
+             be capitalised
+
+The general pattern is:
+
+    [stack]-[app]-[STAGE]
+
+(Clauses separated by hyphens, and the stage capitalised.)
+
+In certain cases, further information may be appended onto the end of
+this pattern as appropriate. For example, a Cloudwatch alarm might be
+called `discussion-search-PROD-disk-util`.
+
+### S3
+
+S3 names are global. As a result, they should be prefaced with
+`[org]-[stack]`. For the Guardian, this might look like:
+`com-gu-discussion`.
+
+Note, '-' is preferred to '.' as a delimiter for bucket names, because
+of SSL issues (see:
+https://wblinks.com/notes/aws-tips-i-wish-id-known-before-i-started/).
+
+### Tags
+
+Each EC2 instance MUST have the following tags:
```

Comment:
Not just EC2 instances. I would say every single AWS resource (EC2 instances, ELBs, LaunchConfigs, VPCs, etc.) MUST have the appropriate tags. Depending on the resource this might be: 

* only `Stage`
* or `Stage` and `Stack`
* or `Stage`, `Stack` and `App`